packages/edit-site/src/components/sidebar/index.js

```diff
@@ -4,31 +4,57 @@
 import { createSlotFill } from '@wordpress/components';
 import { ComplementaryArea } from '@wordpress/interface';
 import { __ } from '@wordpress/i18n';
-import { cog, pencil } from '@wordpress/icons';
+import { cog, typography } from '@wordpress/icons';
+import { GlobalStylesControls } from '@wordpress/global-styles';
+import { useSelect } from '@wordpress/data';
 
 const { Slot: InspectorSlot, Fill: InspectorFill } = createSlotFill(
 	'EditSiteSidebarInspector'
 );
 function Sidebar() {
+	const hasGlobalStylesSupport = useSelect( ( select ) => {
+		return Object.keys( select( 'core/block-editor' ).getSettings() ).some(
+			( key ) => key === '__experimentalGlobalStylesUserEntityId'
+		);
+	} );
 	return (
 		<>
-			<ComplementaryArea.Slot scope="core/edit-site" />
-			<ComplementaryArea
+			<ComplementaryArea.Slot
 				scope="core/edit-site"
-				complementaryAreaIdentifier="edit-site/block-inspector"
-				title={ __( 'Block Inspector' ) }
-				icon={ cog }
-			>
-				<InspectorSlot bubblesVirtually />
-			</ComplementaryArea>
-			<ComplementaryArea
-				scope="core/edit-site"
-				complementaryAreaIdentifier="edit-site/global-styles"
-				title={ __( 'Global Styles' ) }
-				icon={ pencil }
-			>
-				<p>Global Styles area</p>
-			</ComplementaryArea>
+				className="edit-site-complementary-area"
+			/>
+			{ hasGlobalStylesSupport ? (
+				<>
+					<ComplementaryArea
+						scope="core/edit-site"
+						complementaryAreaIdentifier="edit-site/global-styles"
+						title={ __( 'Global Styles' ) }
+						icon={ typography }
+						className="edit-site-complementary-area"
+					>
+						<GlobalStylesControls bubblesVirtually />
+					</ComplementaryArea>
+					<ComplementaryArea
+						scope="core/edit-site"
+						complementaryAreaIdentifier="edit-site/block-inspector"
+						title={ __( 'Block Inspector' ) }
+						icon={ cog }
+						className="edit-site-complementary-area"
+					>
+						<InspectorSlot bubblesVirtually />
+					</ComplementaryArea>
+				</>
+			) : (
+				<ComplementaryArea
```

Comment:
I guess we don't need to repeat the edit-site/block-inspector area code can we unconditionally render it and then only render "edit-site/global-styles" are if hasGlobalStylesSupport is true?

Reply:
I hoped so! Unfortunately, I couldn't make it work due to a weird interaction of block settings & the complementary area slot. This is what happens when we do that:

- block settings for global styles are initially false => the slot renders the inspector control
- block settings for global styles are updated to true => the global styles area is **appended** to the existing areas, so it ends up in the last position.

Reply:
Ah got it, this happens because the order of the fills is not relevant to the order in which they are rendered.
In that case, I suggest, defining a variable with the inspector component to avoid reporting its declaration:
```
const blockInspector = ( <ComplementaryArea
				scope="core/edit-site"
				complementaryAreaIdentifier="edit-site/block-inspector"
				title={ __( 'Block Inspector' ) }
				icon={ cog }
				className="edit-site-complementary-area"
			>
				<InspectorSlot bubblesVirtually />
			</ComplementaryArea>
)
```